sklearn/ensemble/_forest.py

```diff
@@ -815,18 +899,55 @@ def _set_oob_score_and_attributes(self, X, y, scoring_function=None):
         scoring_function : callable, default=None
             Scoring function for OOB score. Defaults to `accuracy_score`.
         """
-        self.oob_decision_function_ = super()._compute_oob_predictions(X, y)
-        if self.oob_decision_function_.shape[-1] == 1:
-            # drop the n_outputs axis if there is a single output
-            self.oob_decision_function_ = self.oob_decision_function_.squeeze(axis=-1)
 
         if scoring_function is None:
             scoring_function = accuracy_score
 
+        ufi_feature_importances, self.oob_decision_function_ = (
+            self._compute_unbiased_feature_importance_and_oob_predictions(
+                X, y, method="ufi"
+            )
+        )
+        mdi_oob_feature_importances, _ = (
+            self._compute_unbiased_feature_importance_and_oob_predictions(
+                X, y, method="mdi_oob"
+            )
+        )
+        if self.criterion == "gini":
+            self._ufi_feature_importances_ = ufi_feature_importances
+            self._mdi_oob_feature_importances_ = mdi_oob_feature_importances
```

Comment:
The Cython code seems to support `criterion="log_loss"` but the result are not exposed as an estimator attribute here.